internal/handshake/crypto_setup.go

```diff
@@ -113,7 +113,8 @@ type cryptoSetup struct {
 
 	zeroRTTParameters      *wire.TransportParameters
 	clientHelloWritten     bool
-	clientHelloWrittenChan chan *wire.TransportParameters
+	clientHelloWrittenChan chan struct{} // is closed as soon as the ClientHello is written
```

Comment:
Is there some reasons to change the type from `chan *wire.TransportParameters` from `chan struct{}`?

Reply:
It was not changed. The channel was renamed to `zeroRTTParametersChan`, and a new `clientHelloWrittenChan` was added.

Reply:
> It was not changed. The channel was renamed to `zeroRTTParametersChan`, and a new `clientHelloWrittenChan` was added.

Ohh, thanks for pointing it out.